ladybug/analysisperiod.py

```diff
@@ -463,3 +452,16 @@ def __repr__(self):
              self.end_time.month, self.end_time.day,
              self.st_time.hour, self.end_time.hour,
              self.timestep)
+
+    def __key(self):
+        return(self.st_time, self.end_time, self.timestep, self.is_leap_year)
```

Comment:
I think you need to `hash()` the datetime objects in order for the `__hash__` method to work correctly. We'd know for sure if you add a test like this:
```
a_period = AnalysisPeriod()
hash(a_period )
```
And you get an exception.